proposals/csharp-11.0/low-level-struct-improvements.md

```diff
@@ -1810,47 +1810,27 @@ void Usage()
 
 To make these APIs usable the compiler ensures that the `ref` lifetime for a `ref` parameter is smaller than lifetime of any references in the associated parameter value. This is the rationale for having *ref-safe-context* for `ref` to `ref struct` be *return-only* and `out` be *caller-context*. That prevents cyclic assignment because of the difference in lifetimes.
 
-It is also why `[UnscopedRef]` only promotes the *ref-safe-context* of any `ref` to `ref struct` values to *return-only* and not *caller-context*. Consider that using *caller-context* allows for cyclic assignment and would force a viral use of `[UnscopedRef]` for a `ref struct`:
```

Comment:
> It is also why `[UnscopedRef]` only promotes the *ref-safe-context* of any `ref` to `ref struct` values to *return-only* and not *caller-context*.

This was incorrect, I believe. Replaced with an `[UnscopedRef]` self-assignment example.

Reply:
Pretty sure this was correct. The core problem once you allow the `ref` and the `ref struct` of a parameter to have the same context is it effectively mandates that the corresponding argument be scoped. It means there is really no way to call that method on a non-scoped value. 

This is particularly problematic when you apply this to `this` (which is another parameter) because it means every instance method call is potentially a cyclic assignment. 

```csharp

void Test()
{
  S local = new();

  // Error: local could now contain a ref to the stack
  local.M1();

  // Error: local could now contain a ref to the stack
  S.M2(ref local);
}

ref struct S { 
  [UnscopedRef]
  void M1() { }

  static void M2([UnscopedRef] ref S p) { }
}
```

Reply:
Think we should change this section to have a list that describes what effect `[UnscopedRef]` has on all the different ref locations:

| ref Location | Original Context | New Context |
| --- | --- | --- |
| instance member | function-member | return-only |
| in / ref | return-only | caller-context |
| out | function-member | return-only |

Essentially a list to that effect.

Reply:
> change this section to have a list that describes what effect `[UnscopedRef]` has

This section is about the silly cyclic assignment, it feels like a better place for such table is in the `#rules-unscoped` section above.

Reply:
That works for me. Just want it somewhere in the doc.